File: belief_state.py
class BeliefState:
    def __init__(self, topic, stance, confidence=0.8):
        self.topic = topic
        self.stance = stance
        self.confidence = confidence
    
    def consider_update(self, proof_score, candidate_stance, threshold=0.6):
        if proof_score < threshold:
            return False
        if candidate_stance.strip().lower() == self.stance.strip().lower():
            return False

        self.stance = candidate_stance
        self.confidence = 0.8
        return True
    
    def hold_firm(self):
        self.confidence = min(0.99, self.confidence + 0.05)

File: web/lib/belief_state.js
// Port of belief_state.py

export class BeliefState {
  constructor(topic, stance, confidence = 0.8) {
    this.topic = topic;
    this.stance = stance;
    this.confidence = confidence;
  }

  considerUpdate(proofScore, candidateStance, threshold = 0.6) {
    if (proofScore < threshold) return false;
    if (candidateStance.trim().toLowerCase() === this.stance.trim().toLowerCase()) return false;

    this.stance = candidateStance;
    this.confidence = 0.8;
    return true;
  }

  holdFirm() {
    this.confidence = Math.min(0.99, this.confidence + 0.05);
  }
}
